shelLF/cli.go

```diff
@@ -0,0 +1,105 @@
+package shelLF
+
+import (
+	"flag"
+	"log"
+	"os"
+	"time"
+)
+
+func NewCLI(toolName string) *CLI {
+	commandMap := make(map[string]*Command)
+	return &CLI{toolName: toolName, commands: commandMap}
+}
+
+func (cli *CLI) AddCommand(cmd Command) {
+	cli.commands[cmd.Name] = &cmd
+}
+
+func (cli *CLI) printUsage() {
+	log.Printf("Usage: %s <command> [options]", cli.toolName)
+	log.Println("Available commands:")
+	for _, cmd := range cli.commands {
+		log.Printf("  %s\t%s\n", cmd.Name, cmd.Description)
```

Comment:
can we format in a better way :

what's my command : %v ,  --> Name 
what I do: %v. -> Description: %v,
what I accept: %v